From: Olaf Hering <olh@suse.de>
Subject: alsa tumbler audio-gpio-active-state handling broken after 2.6.12-rc2

this patch 03a53d4895d274ee8562ddc99723b98d8d0ef04c breaks tumbler
[PATCH] pmac: sound support for latest laptops

audio-gpio-active-state is zero. active_state was forced to 1 in this
case, now it is 0. As a result, no sound is played on my G4/466 and
600Mhz ibook.


This patch fixes it for me.


 sound/ppc/tumbler.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.13-rc7-tumbler/sound/ppc/tumbler.c
===================================================================
--- linux-2.6.13-rc7-tumbler.orig/sound/ppc/tumbler.c
+++ linux-2.6.13-rc7-tumbler/sound/ppc/tumbler.c
@@ -1073,7 +1073,7 @@ static long tumbler_find_device(const ch
 		gp->inactive_val = (*base) ? 0x4 : 0x5;
 	} else {
 		u32 *prop = NULL;
-		gp->active_state = 0;
+		gp->active_state = 1;
 		gp->active_val = 0x4;
 		gp->inactive_val = 0x5;
 		/* Here are some crude hacks to extract the GPIO polarity and
